Bottom: 42c7973398e602e65e87b8abf5450c440eaf08f4
Top:    28bae10b27a2f17de7ce02ad2c043830f3b596c5
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-07-28 11:46:58 +0800

Refresh of tun-xdp-support

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 13abe23..5debe1b 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1027,15 +1027,9 @@ static int tun_xdp_set(struct net_device *dev, struct bpf_prog *prog,
 	 */
 
 	old_prog = rtnl_dereference(tun->xdp_prog);
+	rcu_assign_pointer(tun->xdp_prog, prog);
 	if (old_prog)
 		bpf_prog_put(old_prog);
-	rcu_assign_pointer(tun->xdp_prog, prog);
-
-	if (prog) {
-		prog = bpf_prog_add(prog, 1);
-		if (IS_ERR(prog))
-			return PTR_ERR(prog);
-	}
 
 	return 0;
 }
